### PR TITLE
feat(warn): throws if history is missing

### LINKED ISSUE #843: Not setting history option in initial config leads to confusing error
### What problem does this feature solve?
I'm trying to build my first Vue 3 project, and i'd like to include the new router as well. One thing i noticed is that you get an error that's a bit confusing when not providing the `history` option in the initial configuration, e.g.:

```
const router = createRouter({
    routes : [{
        path : '/',
        component : ScreenHome
    }]
});
```

Leads to an error:

`Uncaught TypeError: Cannot read property 'location' of undefined`

In the console. When i added the `createWebHashHistory` option everything worked fine. 

I know this is mentioned in the docs, but it's easy to miss, and coming from the Vue2 router (where history is set to hash by default) i think this will probably trip up many people trying to convert existing projects or starting new ones.

### What does the proposed API look like?
I think there would be three possible solutions:
1. Set a default `history` mode (like in v3)
2. Make the `history` option required
3. Provide an error message that's clearer ('Please provide a history option')

I would be really happy with either 1) or 2), but at least a better error message would be very helpful.

### PR DESCRIPTION
**Pull request by cheng-github**
Fix #843

### CODE REVIEW

File: src/router.ts [hunk 1]
```diff
@@ -343,6 +343,11 @@ export function createRouter(options: RouterOptions): Router {
   let parseQuery = options.parseQuery || originalParseQuery
   let stringifyQuery = options.stringifyQuery || originalStringifyQuery
   let routerHistory = options.history
+  if (__DEV__ && !routerHistory)
+    throw new Error(
+      'Missing required history option of RouterOptions.\n' +
+        'See more at https://next.router.vuejs.org/api/#createrouter'
```
Comment (posva) [inline src/router.ts:349]:
```suggestion
      'Provide the "history" option when calling "createRouter()":' +
        ' https://next.router.vuejs.org/api/#history.'
```

File: __tests__/router.spec.ts [hunk 2]
```diff
@@ -93,6 +93,11 @@ describe('Router', () => {
     createDom()
   })
 
+  it('fails if history option is missing', () => {
+    // @ts-ignore
+    expect(() => createRouter({ routes })).toThrow()
```
Comment (posva) [inline __tests__/router.spec.ts:98]:
```suggestion
    expect(() => createRouter({ routes })).toThrowError('Provide the "history" option')
```

**Review by posva — CHANGES_REQUESTED** [review]
Thanks!

**Comment by hay** [comment]
Tested and works as expected. Cheers.

**Pull request merged**